Tools/clinic/clinic.py

```diff
@@ -516,11 +516,11 @@ def assert_only_one(
             fcf.format(line)
             def local_fail(should_be_there_but_isnt: bool) -> None:
                 if should_be_there_but_isnt:
-                    fail("{} {} must contain {{{}}} exactly once!".format(
-                        self.__class__.__name__, attr, name))
+                    fail("{cls} {attr} must contain {{{name}}} exactly once!".format(
+                        cls=self.__class__.__name__, attr=attr, name=name))
                 else:
-                    fail("{} {} must not contain {{{}}}!".format(
-                        self.__class__.__name__, attr, name))
+                    fail("{cls} {attr} must not contain {{{name}}}!".format(
+                        cls=self.__class__.__name__, attr=attr, name=name))
```

Comment:
How about doing something like this? (Can't make it a "suggested change" as the range spans deleted lines):

```py
            def local_fail(should_be_there_but_isnt: bool) -> None:
                cls_name = self.__class__.__name__
                if should_be_there_but_isnt:
                    fail(f"{cls_name} {attr!r} must contain {{{name}}} exactly once!")
                else:
                    fail(f"{cls_name} {attr!r} must not contain {{{name}}}!")
```

Reply:
Yep, that's better. I'll get to it.

Reply:
I'm unable to produce coverage for these. I wonder if they are even possible to trigger. Let's revert these too.